project1/client/client_tasks/download_task.cpp

```diff
@@ -1,6 +1,12 @@
 #include "download_task.h"
 
 namespace client_tasks {
+DownloadTask::DownloadTask(const std::string &filename, size_t buf_size,
```

Comment:
I take it you decided not to attempt an initialization list?

For reference, that would look like
```c++
DownloadTask::DownloadTask(const std::string &filename, size_t buf_size, 
                                                  int client_fd) : filename_(filename), buffer_size_(buf_size),
                                                                        client_fd_(client_fd) {}
```

I apologize for the weird formatting; I have to do it by hand.

Reply:
Oh I see. I didn't catch that before. That is a convenient thing to use. I am not sure of the best way to style it, but we can figure that out later.